clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -547,6 +552,17 @@ private Map<String, String> extractPotentialVariables(Map<?, ?> configMap) {
         return new ResolvingMap<>(resolvedOriginals, originals);
     }
 
+    private Predicate<String> automaticConfigProvidersFilter() {
+        String systemProperty = System.getProperty(AUTOMATIC_CONFIG_PROVIDERS_PROPERTY);
+        if (systemProperty == null) {
+            return ignored -> true;
+        } else {
+            return Arrays.stream(systemProperty.split(","))
```

Comment:
Nit: can/should we be more permissive in the parsing logic, to align with how list types are parsed by the AbstractConfig class?

```suggestion
            return Arrays.stream(systemProperty.split(" *, *"))
```

Reply:
The next line trims the whitespace from the strings, so we should already be permissive.

I see that ConfigDef uses `"\\s*,\\s*"` for List properties, while AbstractConfig uses `","`  _without_ trim for `config.providers`. That's not great, but that's also not affected by this PR.

Since `ConfigDef.COMMA_WITH_WHITESPACE` is private, I think i'll leave this as-is.

Reply:
Also split(",") -> trim is more permissive than split(" *, *") or similar, because it removes whitespace at the beginning of the first element and end of the last element. Trim cannot be eliminated in all cases.

Reply:
🤦‍♂️ This is what I get for reviewing on mobile.